### ISSUE TITLE
LDAP/External Authentication Support

### ISSUE THREAD

**Issue by Diablofan** [ciur/papermerge#94]
I was wondering if external authentication support, like LDAP, could be added in.

It would be nice if it were supported as it would allow me to delegate out user logins without having to set them up in the app and make sure they are constantly in sync (say with password changes).

It would also allow me to use the same authentication mechanism that all the other applications I have installed on my server are also using.

I'm sure this also would be beneficial for larger orgs since they would probably want some sort of directory based authentication like LDAP or Active Directory, again, for the same reasons I listed above.

**Comment by ciur** [comment]
Will be part of 1.6.0 release.

**Comment by ramon-kramer** [comment]
> Planned as part of 1.6.0 release.

hi, any news about this? i did not find it in changelog, so when will it be available?

**Comment by ciur** [comment]
> is there any chance to get this forward like donation?

@ramon-kramer, sure. #231 details on different ways to support the project.

Please, keep in mind that support for LDAP authentication will be implemented as part of 2.2.0 version which will be release following year (i.e. 2022).

**Comment by phinnay** [comment]
Bumping the thread now that it's 2022, is there an ETA for this feature and will donations help? Loving Papermerge!!!

**Comment by ciur** [comment]
@phinnay, donations are welcome, but they won't speed development :smile: 

Once you brought up the topic, let me say couple of words here. I am in chatty mode :smile:

Last stable release was out exactly one year ago. So, it makes perfect sense to ask, why releasing next version takes more than a year (in this agile world, where two weeks release cycle is norm) ?

It is because I have long term plans for Papermerge. In order for software to be long term sustainable it needs a solid architectural foundation. However, version 2.0, had serious design issues. Just to mention one problem - the frontend code -  was ever growing mess... for relatively young application, javascript code was seriously problematic.
I instead of continuing applying minor design fixes here and there - I decided to completely rewrite frontend part.

It took me a lot of time to figure out how to get done the frontend part "right way". "Right" code is application code which is relatively well organized, and regardless how much application increases in complexity - code is still easy to read and make sense of.

Backend code had serious changes but in overall did not change much.

In short, it will take some time until 2.1, and it will take even more time until LDAP authentication will be out (as part of 2.2).
Building quality applications needs time and patients and I assure you it is worth it.

Off topic, but I cannot help and not to show you what I am playing with currently:



![quick-preview-work-in-progress](https://user-images.githubusercontent.com/24827601/162375678-5c771982-9fa6-463a-b286-fa70fdb3f9b1.gif)

**Comment by ciur** [comment]
Papermerge 3.2 has OIDC (OpenID connect) support. I've tested it with keycloak and it works. Keycloak has user federation i.e. loading your users from ldap source (to be exact sync with ldap). What you ca do, is configure papermerge with keycloak and let keyloak handle ldap part.

**Comment by ciur** [comment]
Papermerge 3.2 has OIDC (OpenID connect) support. I've tested it with keycloak and it works. Keycloak has user federation i.e. loading your users from ldap source (to be exact sync with ldap). What you ca do, is configure papermerge with keycloak and let keyloak handle ldap part. 

I am closing this ticket as feature is already landed in 3.2 (via keycloak/OIDC support).
